gbpservice/neutron/tests/unit/nfp/configurator/drivers/vpn/vyos/test_vpn_driver.py

```diff
@@ -0,0 +1,424 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+from gbpservice.nfp.configurator.agents import vpn
+from gbpservice.nfp.configurator.drivers.vpn.vyos import vyos_vpn_driver
+from gbpservice.neutron.tests.unit.nfp.configurator.test_data import (
+                                                                test_vpn_data)
+import json
+import unittest
+import mock
+import requests
+
+""" Implements test cases for driver methods
+of vpn.
+
+
+"""
+
+
+class VpnaasIpsecDriverTestCase(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(VpnaasIpsecDriverTestCase, self).__init__(*args, **kwargs)
+        self.dict_objects = test_vpn_data.MakeDictionaries()
+        self.context = self.dict_objects._make_service_context()
+        self.plugin_rpc = vpn.VpnaasRpcSender(self.context,
+                                              self.dict_objects.sc)
+        self.driver = vyos_vpn_driver.VpnaasIpsecDriver(self.plugin_rpc)
+        self.svc_validate = vyos_vpn_driver.VPNSvcValidator(self.plugin_rpc)
+        self.resp = mock.Mock()
+        self.fake_resp_dict = {'status': True}
+
+    def test_create_vpn_service(self):
+        '''
+        Implements method to test the vpn driver's create vpn service.
+        '''
+
+        context = self.dict_objects._make_service_context(operation_type='vpn')
+
+        kwargs = self.dict_objects._make_kwargs(operation='create',
+                                                service_type='vpn')
+        with mock.patch.object(self.plugin_rpc, 'update_status') as (
+                                                mock_update_status):
+            self.driver.vpnservice_updated(context, kwargs)
+            mock_update_status.assert_called_with(
+                                        self.context,
+                                        self.dict_objects.vpn_vpnsvc_active)
+
+    def test_create_ipsec_site_conn(self):
```

Comment:
Looks like put method is unused in VPN driver. Didn't pep8/flake8 complain about this?